src/parkapi_sources/converters/herrenberg_bike/converter.py

```diff
@@ -0,0 +1,86 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+from abc import ABC
+
+import requests
+from validataclass.exceptions import ValidationError
+from validataclass.validators import DataclassValidator
+
+from parkapi_sources.converters.base_converter.pull import GeojsonInput, PullConverter
+from parkapi_sources.exceptions import ImportParkingSiteException, ImportSourceException
+from parkapi_sources.models import RealtimeParkingSiteInput, SourceInfo, StaticParkingSiteInput
+
+from .models import HerrenbergBikeFeatureInput
+
+
+class HerrenbergBikePullConverter(PullConverter, ABC):
+    geojson_validator = DataclassValidator(GeojsonInput)
+    herrenberg_feature_validator = DataclassValidator(HerrenbergBikeFeatureInput)
+
+    source_info = SourceInfo(
+        uid='herrenberg_bike',
+        name='Stadt Herrenberg - Munigrid: Fahrrad-Abstellanlagen',
+        public_url='https://www.munigrid.de/hbg/dataset/8',
+        source_url='https://www.munigrid.de/api/file/11/latest',
+        timezone='Europe/Berlin',
+        attribution_contributor='Stadt Herrenberg - Munigrid',
+        attribution_license='CC 0 1.0',
+        attribution_url='https://creativecommons.org/publicdomain/zero/1.0/',
+        has_realtime_data=False,
+    )
+
+    def _get_feature_inputs(self) -> tuple[list[HerrenbergBikeFeatureInput], list[ImportParkingSiteException]]:
+        feature_inputs: list[HerrenbergBikeFeatureInput] = []
+        import_parking_site_exceptions: list[ImportParkingSiteException] = []
+
+        response = requests.get(self.source_info.source_url, timeout=30)
+        response_data = response.json()
+
+        try:
+            geojson_input = self.geojson_validator.validate(response_data)
+        except ValidationError as e:
+            raise ImportSourceException(
+                source_uid=self.source_info.uid,
+                message=f'Invalid Input at source {self.source_info.uid}: {e.to_dict()}, data: {response_data}',
+            ) from e
+
+        for feature_dict in geojson_input.features:
+            if self._should_ignore_dataset(feature_dict):
+                continue
+
+            try:
+                feature_input = self.herrenberg_feature_validator.validate(feature_dict)
+            except ValidationError as e:
+                import_parking_site_exceptions.append(
+                    ImportParkingSiteException(
+                        source_uid=self.source_info.uid,
+                        parking_site_uid=feature_dict.get('properties').get('id'),
+                        message=f'Invalid data at uid {feature_dict.get("properties").get("id")}: ' f'{e.to_dict()}, data: {feature_dict}',
```

Comment:
This will throw an exception if `properties` is not set. Please set `{}` as default

Reply:
It's now set as default